pymongo/errors.py

```diff
@@ -61,6 +61,15 @@ class ConnectionFailure(PyMongoError):
     """Raised when a connection to the database cannot be made or is lost."""
 
 
+class WaitQueueTimeoutError(ConnectionFailure):
+    """Raised when we time out due to the query wait time.
```

Comment:
"Raised when an operation times out waiting to checkout a connection from the pool."

Reply:
Fixed